### PR TITLE
Update confirmation titles to be more descriptive

### LINKED ISSUE #313: Confirmation messages should be unique and more detailed about what's exactly affected
The current confirmation messages are too generic

screenshot 2016-03-31 14 48 45

In this example, the title should indicate what's being done exactly (Retry messages) and which items are affected exactly (number of messages)

CC // @mauroservienti

### PR DESCRIPTION
**Pull request by WojcikMike**
Fixes #313 

The confirmation messages need to be reviewed

### CODE REVIEW

File: src/ServicePulse.Host/app/js/views/archive/view.html [hunk 1]
```diff
@@ -11,7 +11,7 @@
         <div class="row" ng-show="!vm.loadingData">
             <div class="col-sm-12">
                 <div class="btn-toolbar">
-                    <button type="button" class="btn btn-default" confirm-message="Are you sure you want to un-archive selected archived messages?" confirm-click="vm.unarchiveSelected()" ng-disabled="vm.selectedIds.length == 0"><i class="fa fa-undo"></i> Unarchive {{vm.selectedIds.length | number}} selected</button>
+                    <button type="button" class="btn btn-default" confirm-title="Are you sure you want to un-archive selected archived messages" confirm-message="Are you sure you want to un-archive selected archived messages?" confirm-click="vm.unarchiveSelected()" ng-disabled="vm.selectedIds.length == 0"><i class="fa fa-undo"></i> Unarchive {{vm.selectedIds.length | number}} selected</button>
```
Comment (WojcikMike) [inline src/ServicePulse.Host/app/js/views/archive/view.html]: @sergioc What kind of message should be here? The confirmation message is a 1 line only so repeating it in title and message looks bad.

File: src/ServicePulse.Host/app/js/views/failed_groups/view.html [hunk 2]
```diff
@@ -56,10 +56,10 @@
                                         <button type="button" class="btn btn-link btn-sm hidden-sm-down" title="View messages in this Group" ng-click="vm.viewExceptionGroup(group)" ng-disabled="group.count == 0">
                                             View Messages
                                         </button>
-                                        <button type="button" class="btn btn-link btn-sm" title="Retry messages in this Group" confirm-click="vm.retryExceptionGroup(group)" ng-disabled="group.count == 0 || group.workflow_state.status !== 'ready'" confirm-message="Are you sure you want to retry this group of {{group.count}} messages?">
+                                        <button type="button" class="btn btn-link btn-sm" title="Retry messages in this Group" confirm-click="vm.retryExceptionGroup(group)" ng-disabled="group.count == 0 || group.workflow_state.status !== 'ready'" confirm-title="Are you sure you want to retry this group?" confirm-message="Are you sure you want to retry this group of {{group.count}} messages?">
```
Comment (WojcikMike) [inline src/ServicePulse.Host/app/js/views/failed_groups/view.html]: As above

File: src/ServicePulse.Host/app/js/views/failed_groups/view.html [hunk 3]
```diff
@@ -56,10 +56,10 @@
...
                                             Retry Group
                                         </button>
-                                        <button type="button" class="btn btn-link btn-sm" title="Archive messages in this Group" confirm-click="vm.archiveExceptionGroup(group)" ng-disabled="group.count == 0 || group.workflow_state.status !== 'ready'" confirm-message="Are you sure you want to archive this group of messages? If you do, they will not be available for Retry.">
+                                        <button type="button" class="btn btn-link btn-sm" title="Archive messages in this Group" confirm-click="vm.archiveExceptionGroup(group)" ng-disabled="group.count == 0 || group.workflow_state.status !== 'ready'" confirm-title="Are you sure you want to archive this group?" confirm-message="Are you sure you want to archive this group of messages? If you do, they will not be available for Retry.">
```
Comment (WojcikMike) [inline src/ServicePulse.Host/app/js/views/failed_groups/view.html]: As above

File: src/ServicePulse.Host/app/js/views/failed_messages/view.html [hunk 4]
```diff
@@ -27,9 +27,9 @@
...
+                    <button type="button" class="btn btn-default" confirm-title="Are you sure you want to archive the selected messages?" confirm-message="If you do they will not be available for Retry." confirm-click="vm.archiveSelected()" ng-disabled="vm.selectedIds.length == 0"><i class="fa fa-archive"></i> Archive {{vm.selectedIds.length | number}} selected</button>
+                    <button type="button" class="btn btn-default" confirm-title="Are you sure you want to retry the whole group?" confirm-message="Retrying a whole group can take some time and put extra load on your system. Are you sure you want to retry all these messages?" confirm-click="vm.retryExceptionGroup(vm.selectedExceptionGroup)"><i class="fa fa-refresh"></i> Retry all</button>
+                    <button type="button" class="btn btn-default" ng-show="vm.selectedExceptionGroup.id" confirm-title="Are you really sure you want to archive this group?" confirm-message="Are you really sure you want to archive this group?" confirm-click="vm.archiveExceptionGroup(vm.selectedExceptionGroup)"><i class="fa fa-archive"></i> Archive all</button>
```
Comment (WojcikMike) [inline src/ServicePulse.Host/app/js/views/failed_messages/view.html]: And this

File: src/ServicePulse.Host/app/js/views/redirect/view.html [hunk 5]
```diff
@@ -48,7 +48,7 @@
                         <div isolate-click class="col-sm-12">
                             <p class="small">
 
-                                <button type="button" class="btn btn-link btn-sm" confirm-title="Confirm action" confirm-click="vm.deleteRedirect(redirect, 'Redirect was deleted', 'Failed to delete redirect')" 
+                                <button type="button" class="btn btn-link btn-sm" confirm-title="Are you sure you want to end the redirect?" confirm-click="vm.deleteRedirect(redirect, 'Redirect was deleted', 'Failed to delete redirect')"
```
Comment (WojcikMike) [inline src/ServicePulse.Host/app/js/views/redirect/view.html]: And this :)

File: src/ServicePulse.Host/app/js/views/archive/view.html (hunk 1 above)
Comment (kbaley) [inline src/ServicePulse.Host/app/js/views/archive/view.html]: If the existing message stays, it needs a ? at the end.

File: src/ServicePulse.Host/app/js/views/failed_messages/view.html (hunk 4 above)
Comment (kbaley) [inline src/ServicePulse.Host/app/js/views/failed_messages/view.html]: > Are you really sure you want to archive this group?

Not a fan of saying "really" in this message (or the other ones). Do we have two levels of confirmation that require it?

File: src/ServicePulse.Host/app/js/views/archive/view.html (hunk 1 above)
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/archive/view.html]: Are there any consequences to un-archiving messages?

File: src/ServicePulse.Host/app/js/views/failed_groups/view.html (hunk 2 above)
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/failed_groups/view.html]: Any special notes, concerns or consequences the user should be aware of before deciding to retry?

File: src/ServicePulse.Host/app/js/views/failed_groups/view.html (hunk 3 above)
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/failed_groups/view.html]: Any special notes, concerns or consequences the user should be aware of before deciding to archive?

File: src/ServicePulse.Host/app/js/views/failed_messages/view.html (hunk 4 above)
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/failed_messages/view.html]: Any special notes, concerns or consequences the user should be aware of before deciding to archive groups?

File: src/ServicePulse.Host/app/js/views/redirect/view.html (hunk 5 above)
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/redirect/view.html]: Any special notes, concerns or consequences the user should be aware of before deciding to end the redirect?

**Comment by WojcikMike** [comment]
@kbaley @sergioc I addressed your feedback, please have a look

File: src/ServicePulse.Host/app/js/views/archive/view.html [hunk 6]
```diff
@@ -11,7 +11,7 @@
         <div class="row" ng-show="!vm.loadingData">
             <div class="col-sm-12">
                 <div class="btn-toolbar">
-                    <button type="button" class="btn btn-default" confirm-message="Are you sure you want to un-archive selected archived messages?" confirm-click="vm.unarchiveSelected()" ng-disabled="vm.selectedIds.length == 0"><i class="fa fa-undo"></i> Unarchive {{vm.selectedIds.length | number}} selected</button>
+                    <button type="button" class="btn btn-default" confirm-title="Are you sure you want to un-archive selected archived messages?" confirm-message="The messages will be visible on the failed message screen." confirm-click="vm.unarchiveSelected()" ng-disabled="vm.selectedIds.length == 0"><i class="fa fa-undo"></i> Unarchive {{vm.selectedIds.length | number}} selected</button>
```
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/archive/view.html]: `Unarchived messages will be moved back to the list of failed messages.`

@WojcikMike in addition to this, are there any potential consequences for queues, endpoints or the system in general?

File: src/ServicePulse.Host/app/js/views/failed_messages/view.html [hunk 7]
```diff
@@ -27,9 +27,9 @@
...
-                    <button type="button" class="btn btn-default" confirm-message="Retrying a whole group can take some time and put extra load on your system. Are you sure you want to retry all these messages?" confirm-click="vm.retryExceptionGroup(vm.selectedExceptionGroup)"><i class="fa fa-refresh"></i> Retry all</button>
-                    <button type="button" class="btn btn-default" ng-show="vm.selectedExceptionGroup.id" confirm-message="Are you really sure you want to archive this group?" confirm-click="vm.archiveExceptionGroup(vm.selectedExceptionGroup)"><i class="fa fa-archive"></i> Archive all</button>
+                    <button type="button" class="btn btn-default" confirm-title="Are you sure you want to archive the selected messages?" confirm-message="If you do they will not be available for Retry." confirm-click="vm.archiveSelected()" ng-disabled="vm.selectedIds.length == 0"><i class="fa fa-archive"></i> Archive {{vm.selectedIds.length | number}} selected</button>
```
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/failed_messages/view.html]: `If you archive, these messages won't be available for retrying unless they're later unarchived.`

File: src/ServicePulse.Host/app/js/views/failed_messages/view.html [hunk 8]
```diff
@@ -27,9 +27,9 @@
...
+                    <button type="button" class="btn btn-default" confirm-title="Are you sure you want to retry the whole group?" confirm-message="Retrying a whole group can take some time and put extra load on your system. Are you sure you want to retry all these messages?" confirm-click="vm.retryExceptionGroup(vm.selectedExceptionGroup)"><i class="fa fa-refresh"></i> Retry all</button>
+                    <button type="button" class="btn btn-default" ng-show="vm.selectedExceptionGroup.id" confirm-title="Are you sure you want to archive this group?" confirm-message="If you do messages from that group will not be available for Retry." confirm-click="vm.archiveExceptionGroup(vm.selectedExceptionGroup)"><i class="fa fa-archive"></i> Archive all</button>
```
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/failed_messages/view.html]: `If you archive, the messages in the group won't be available for retrying unless they're later unarchived.`

File: src/ServicePulse.Host/app/js/views/redirect/view.html [hunk 9]
```diff
@@ -48,8 +48,8 @@
                         <div isolate-click class="col-sm-12">
                             <p class="small">
 
-                                <button type="button" class="btn btn-link btn-sm" confirm-title="Confirm action" confirm-click="vm.deleteRedirect(redirect, 'Redirect was deleted', 'Failed to delete redirect')" 
-                                        confirm-message="Are you sure you want to end the redirect?">
+                                <button type="button" class="btn btn-link btn-sm" confirm-title="Are you sure you want to end the redirect?" confirm-click="vm.deleteRedirect(redirect, 'Redirect was deleted', 'Failed to delete redirect')" 
+                                        confirm-message="Without this redirect retried messages will be send to its destination queue.">
```
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/redirect/view.html]: `Once the redirect is ended, any affected messages will be sent to the original destination queue. Ensure this queue is ready to accept messages again.`

File: src/ServicePulse.Host/app/js/views/archive/view.html (hunk 6 above)
Comment (WojcikMike) [inline src/ServicePulse.Host/app/js/views/archive/view.html]: @sergioc no, there is not.

**Comment by WojcikMike** [comment]
All changes incorporated @sergioc please merge

File: src/ServicePulse.Host/app/js/views/archive/view.html (hunk 6 above)
Comment (sergioc) [inline src/ServicePulse.Host/app/js/views/archive/view.html]: OK, then I'll stick to my the revision I recommended in the previous comment.

**Pull request merged**